### PR TITLE
prevent a float being passed to str_repeat

### PR DESCRIPTION
**Pull request by paaaaaaaaaa**
## Hot fix
This pull request will prevent php type casting a float when a integer is required. Resulting in the following fatal error.

PHP Fatal error:  Uncaught TypeError: str_repeat() expects parameter 2 to be integer, float given...

This may happen when a custom secret is generated that is not 10 characters long.

### CODE REVIEW

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by paaaaaaaaaa** [comment]
@greg0ire I'm happy to squash commits my end, however I believe you can easily do this when merging the pull request? 
image

**Comment by greg0ire** [comment]
>  If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

That's what I meant by that.

File: tests/FixedBitNotationTest.php [hunk 1]
```diff
@@ -53,6 +53,18 @@ public function testDecodeIsSameAsEncode($input): void
         $this->assertSame((string) $input, $bits->decode($bits->encode($input)));
     }
 
+    /**
+     * Test with padding.
+     *
+     * @dataProvider decodeIsSameAsEncode
+     */
+    public function testDecodeIsSameAsEncodeWithPadding($input): void
```
Comment (greg0ire) [inline tests/FixedBitNotationTest.php]:
```suggestion
    public function testDecodeIsSameAsEncodeWithPadding(string $input): void
```

**Comment by paaaaaaaaaa** [comment]
Created a new pull request with the squashed commits.

**Comment by greg0ire** [comment]
> Created a new pull request with the squashed commits.

You did not have to… were my instructions unclear?

**Pull request closed without merging**